### PR TITLE
Add functionality to download training metrics graphs

### LINKED ISSUE #5656: Downloadable metrics graphs
## Objective

Allow the user to download the training metrics

## Motivation

The user can view training-time metrics such as the loss over time, learning rate schedule, etc... in the UI, but he cannot download the raw data. The option to download the metrics data in a human-readable format can be helpful to users who need it for papers or presentations.

## Requirements

- Allow exporting the training-metrics in a human-readable format (e.g. image or .pdf) from the UI

### PR DESCRIPTION
**Pull request by andersendsa**
### Summary
- Implemented a "Download" button in the `MetricGraph` component to let users export training metrics graphs as PNG images.
- Extracted SVG elements natively, drew them on a canvas, and triggered a PNG download directly, without introducing extra dependencies.

Closes: #5656

### How to test

1. Open the application UI and navigate to your project.
2. Click on the Models tab in the main navigation menu.
3. Select a trained model from the list to open the Model Details view.
4. Click on the Training Metrics tab inside the details panel.
5. You should see one or more line graphs (e.g., Training Loss, Validation Accuracy, etc.).
6. Look at the top-right corner of any graph's container box. You should now see a new Download icon button.
7. Click the Download button.
8. Verify Download: A .png file (named based on the graph's title, e.g., training_loss_metrics.png) should be downloaded to your local machine.
9. Verify Image Quality: Open the downloaded image file. Confirm that the graph is fully visible with a solid white background, and that all data lines, axes, and labels rendered correctly exactly as they appeared in the UI.

### Checklist

- [x] The PR title and description are clear and descriptive
- [x] I have manually tested the changes
- [x] All changes are covered by automated tests
- [x] All related issues are linked to this PR (if applicable)
- [ ] Documentation has been updated (if applicable)

### CODE REVIEW

**Comment by andersendsa** [comment]
Hi @dwesolow, @jpggvilaca, @A-Artemis   pls let me know if this pr needs any changes or if it is good to merge

**Comment by andersendsa** [comment]
Hi @leoll2  could you pls trigger the checks again ?

**Comment by leoll2** [comment]
@andersendsa I tested the feature locally and there are evident problems with the generated image (missing data, poor formatting, ...). However, you say that you tested it and that everything is rendered correctly. Could you maybe share a screenshot of how the graphs look in your environment?

**Comment by andersendsa** [comment]
> @andersendsa I tested the feature locally and there are evident problems with the generated image (missing data, poor formatting, ...). However, you say that you tested it and that everything is rendered correctly. Could you maybe share a screenshot of how the graphs look in your environment?

Hi @leoll2  sorry for the late reply I was busy in my university examinations I hadnt pushed the commit sorry for that here is the graph I am getting 
image

**Comment by leoll2** [comment]
Thanks for applying the fixes. To be honest, I'm a bit hesitant to accept the feature in its current state. While downloading graphs is certainly useful for researchers, the graphs themselves look a bit rough:

- The units of measurement are absent.
- Axis intervals are not aligned with the data points; they mark for example values like "2.8 epochs" which have no physical meaning.
- Decimal precision is inconsistent
- Text overlaps

Data visualization isn't just a matter of style, it is about presenting information in a digestible and credible way. If you are willing to improve the graphs in the aspects that I mentioned above, then I would be happy to eventually merge the PR, at the moment it's not "production-ready" yet.

**Comment by andersendsa** [comment]
> Thanks for applying the fixes. To be honest, I'm a bit hesitant to accept the feature in its current state. While downloading graphs is certainly useful for researchers, the graphs themselves look a bit rough:
> 
> * The units of measurement are absent.
> * Axis intervals are not aligned with the data points; they mark for example values like "2.8 epochs" which have no physical meaning.
> * Decimal precision is inconsistent
> * Text overlaps
> 
> Data visualization isn't just a matter of style, it is about presenting information in a digestible and credible way. If you are willing to improve the graphs in the aspects that I mentioned above, then I would be happy to eventually merge the PR, at the moment it's not "production-ready" yet.

Thank you for the feedback. I’d love to continue working on this PR and keep receiving reviews.

**Comment by andersendsa** [comment]
@leoll2  pls let me know if any more changes are required 
image

**Review by leoll2 — DISMISSED** [review]

**Comment by leoll2** [comment]
Thanks for the improvements, the feature now works as expected 😃

**Comment by leoll2** [comment]
@andersendsa A few Eslint checks are failing and blocking this PR, would you check them?

https://github.com/open-edge-platform/geti/actions/runs/31010971233/job/95329949541?pr=6917

**Comment by andersendsa** [comment]
> @andersendsa A few Eslint checks are failing and blocking this PR, would you check them?
> 
> https://github.com/open-edge-platform/geti/actions/runs/31010971233/job/95329949541?pr=6917

Hi @leoll2  I have fixed the failing checks could you pls trigger  the checks ?

File: application/ui/src/features/models/model-listing/model-metrics/metric-graph.component.tsx [hunk 1]
```diff
@@ -22,37 +25,166 @@ const X_AXIS_TICK_COUNT = 8;
...
+                const link = document.createElement('a');
+                link.href = dataUrl;
+                link.download = `${title.replace(/\s+/g, '_').toLowerCase()}_metrics.png`;
+                link.click();
+            }
+
+            URL.revokeObjectURL(url);
+        };
+        img.src = url;
+    }, [title, unwrappedGraphRef]);
```
Comment (jpggvilaca) [inline application/ui/src/features/models/model-listing/model-metrics/metric-graph.component.tsx:48]: This function has 100 lines and several responsibilities. Could we extract the logic parts out of this to increase readability? Also, is it possible to add a few unit tests for this?

File: application/ui/src/features/models/model-listing/model-metrics/download-graph.utils.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
+/* eslint-disable header/header */
```
Comment (jpggvilaca) [inline application/ui/src/features/models/model-listing/model-metrics/download-graph.utils.ts:1]: why disable header?

**Review by jpggvilaca — APPROVED** [review]

**Review by leoll2 — APPROVED** [review]
